extract.go

```diff
@@ -32,28 +32,67 @@ import (
 	"path/filepath"
 
 	"github.com/juju/errors"
+	"gopkg.in/h2non/filetype.v1"
```

Comment:
This package allows the type to be inferred from header bytes, extension and MIME Type